vaccine_feed_ingest/run.py

```diff
@@ -93,47 +127,163 @@ def _find_executeable(site_dir: pathlib.Path, cmd_name: str) -> Optional[pathlib
     return cmd
 
 
-def _run_fetch(site_dir: pathlib.Path) -> None:
-    fetch_path = _find_executeable(site_dir, FETCH_CMD)
+def _generate_run_timestamp() -> str:
+    return datetime.datetime.now().replace(microsecond=0).isoformat()
+
+
+def _find_all_run_dirs(
+    base_output_dir: pathlib.Path,
+    site: str,
+    stage: PipelineStage,
+) -> Iterator[pathlib.Path]:
+    """Find latest stage output path"""
+    stage_dir = base_output_dir / site / STAGE_OUTPUT_NAME[stage]
+
+    for run_dir in sorted(stage_dir.iterdir(), reverse=True):
+        if run_dir.name.startswith("_"):
+            continue
+
+        if run_dir.name.startswith("."):
+            continue
+
+        yield run_dir
+
+
+def _find_latest_run_dir(
+    base_output_dir: pathlib.Path,
+    site: str,
+    stage: PipelineStage,
+) -> Optional[pathlib.Path]:
+    """Find latest stage output path"""
+    return next(_find_all_run_dirs(base_output_dir, site, stage), None)
+
+
+def _generate_run_dir(
+    base_output_dir: pathlib.Path,
+    site: str,
+    stage: PipelineStage,
+    timestamp: str,
+) -> pathlib.Path:
+    """Generate output path for a pipeline stage."""
+    run_dir = base_output_dir / site / STAGE_OUTPUT_NAME[stage] / timestamp
+
+    run_dir.mkdir(parents=True, exist_ok=True)
+
+    return run_dir
+
+
+def _copy_files(src_dir: pathlib.Path, dst_dir: pathlib.Path) -> None:
+    for filepath in src_dir.iterdir():
+        if filepath.name.startswith("_") or filepath.name.startswith("."):
+            continue
+
+        with filepath.open("rb") as src_file:
+            with (dst_dir / filepath.name).open("wb") as dst_file:
+                for content in src_file:
+                    dst_file.write(content)
+
+
+def _run_fetch(
+    site_dir: pathlib.Path,
+    output_dir: pathlib.Path,
+    timestamp: str,
+) -> None:
+    fetch_path = _find_executeable(site_dir, PipelineStage.FETCH)
     if not fetch_path:
         logger.info("No fetch cmd in %s to run.", str(site_dir))
         return
 
+    fetch_run_dir = _generate_run_dir(
+        output_dir, site_dir.name, PipelineStage.FETCH, timestamp
+    )
+
     with tempfile.TemporaryDirectory(f"_fetch_{site_dir.name}") as tmp_str:
         tmp_dir = pathlib.Path(tmp_str)
+        fetch_output_dir = tmp_dir / "output"
+        fetch_output_dir.mkdir(parents=True, exist_ok=True)
+
+        subprocess.run([str(fetch_path), str(fetch_output_dir)], check=True)
 
-        subprocess.run([str(fetch_path), str(tmp_dir)], check=True)
+        _copy_files(fetch_output_dir, fetch_run_dir)
```

Comment:
Correct. Copy files will copy the files to the local `out` directory with running in development, but when running in production it will copy to GCS.

Depends on what the `--output-dir` is pointing to.